### PR TITLE
20230720 unix int modular

### PR DESCRIPTION
**Pull request by Firstyear**
This makes the Db cache slightly more modular, getting ready for it to be a dyn trait or more portable/exposed. cc @dmulder since this is of interest to you :) 

Next will probably be to make the client it's own trait too.

Checklist

- [ x ] This pr contains no AI generated code
- [ x ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: unix_integration/src/daemon.rs [hunk 1]
```diff
@@ -660,8 +661,16 @@ async fn main() -> ExitCode {
...
-                cfg.db_path.as_str(), // The sqlite db path
+            let db = match Db::new(cfg.db_path.as_str(), &cfg.tpm_policy) {
+                Ok(db) => db,
+                Err(_e) => {
+                    error!("Failed to create database");
+                    return ExitCode::FAILURE
+                }
+            };
+
+            let cl_inner = match Resolver::new(
```
Comment (yaleman) [inline unix_integration/src/daemon.rs:672]: At some point we should probably just pass `db, rsclient, cfg.clone()` and deal with it internally 😄

**Review by yaleman — COMMENTED** [review]
There's a *lot* of `.map_err(|_| ())?` with no logging/feedback in the calls for admins to find out where/why things are failing, this feels bad - but by the looks of it, it's not making it more of an issue than the current situation?

File: unix_integration/src/daemon.rs (hunk 1 above)
Comment (Firstyear) [reply unix_integration/src/daemon.rs:672]: Yes, that's the goal. Sorry there were some emails that @dmulder and I had that you weren't cc'ed on. I'll forward you the goals - docs. But that's the plan :)

**Comment by Firstyear** [comment]
> There's a _lot_ of `.map_err(|_| ())?` with no logging/feedback in the calls for admins to find out where/why things are failing, this feels bad - but by the looks of it, it's not making it more of an issue than the current situation?

Yes, this is going to be improved here too. Here we actually log in the lower level so the map errs here are just to allow bubbling up but the errors are logged at a lower level.

So this is on the todo :)

**Comment by Firstyear** [comment]
Updated cargo.lock, rebased to master.

**Pull request merged**